test/libsolidity/Imports.cpp

```diff
@@ -104,8 +104,7 @@ BOOST_AUTO_TEST_CASE(simple_alias)
 BOOST_AUTO_TEST_CASE(library_name_clash)
 {
 	CompilerStack c;
-	c.addSource("a", "library A {} pragma solidity >=0.0;");
-	c.addSource("b", "library A {} pragma solidity >=0.0;");
+	c.addSource("a", "library A {} library A {} pragma solidity >=0.0;");
```

Comment:
Regardless of what they will be called, this contract will produce an error in the type checker and not in the linker. Can you please also check how the library is called in the unlinked bytecode? There should be something like `__A______` or `__a:A____`.